### PR TITLE
Add Alt-dragging to duplicate layers

### LINKED ISSUE #766: Alt-drag should duplicate layers
Allow duplication by dragging while holding Alt, since we have a keyboard hint that doesn't say "(Coming Soon)" saying this is possible

### PR DESCRIPTION
**Pull request by 0HyperCube**
Closes #766
> Allow duplication by dragging while holding Alt, since we have a keyboard hint that doesn't say "(Coming Soon)" saying this is possible

Notes:
- Duplicated layers go to the top of the folder of the folder that the previously selected layers were in.
- You drag the duplicated layers and not the originals.
- Dragging a layer as well as its parent duplicates just the parent.

### CODE REVIEW

**Comment by Keavon** [comment]
We'd like it so holding <kbd>Alt</kbd>, and then clicking and dragging an unselected layer, should perform the operation. Not only if it's already selected. Thanks :)

**Comment by Keavon** [comment]
It would also be great (although I realize it would be a little harder to implement) if you can randomly add or remove the <kbd>Alt</kbd> key at any time before or during the drag makes it so the layer is duplicated or not duplicated.

- If you're holding the <kbd>Alt</kbd> key at any time: you are dragging the copy
- If you're not holding the <kbd>Alt</kbd> key at any time: you are dragging the original

This does mean that the duplicate layer(s) has to be created or deleted potentially as many times as you press and release the key.

**Comment by 0HyperCube** [comment]
@Keavon These are both added now.

**Comment by Keavon** [comment]
> @Keavon These are both added now.

Nice! This works perfectly. 👌

File: editor/src/messages/tool/tool_messages/select_tool.rs [hunk 1]
```diff
@@ -509,6 +514,12 @@ impl Fsm for SelectToolFsmState {
 						.flat_map(snapping::expand_bounds)
 						.collect();
 
+					if input.keyboard.get(duplicate as usize) && tool_data.not_duplicated_layers.is_none() {
+						tool_data.duplicate(document, responses);
+					} else if !input.keyboard.get(duplicate as usize) && tool_data.not_duplicated_layers.is_some() {
+						tool_data.undo_duplicate(responses);
```
Comment (Keavon) [inline editor/src/messages/tool/tool_messages/select_tool.rs:520]: How about calling them `create_duplicate()` and `remove_duplicate()`?

**Review by Keavon — APPROVED** [review]
I'm surprised this can be implemented with less than 100 lines of code added. It's very clean!

**Pull request merged**